### PR TITLE
LPS-86729 journal-test: Add integration test to check the indexed fields and reindexing for JournalArticle.

### PR DESCRIPTION
**Pull request by brandizzi**

### CODE REVIEW

**Comment by leomiyagi** [comment]
ci:test:search:norebase

**Comment by leomiyagi** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerIndexedFieldsTest.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+ */
+@RunWith(Arquillian.class)
+public class JournalFolderIndexerIndexedFieldsTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE,
```
Comment (leomiyagi) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerIndexedFieldsTest.java]: This needs to be changed to PermissionCheckerMethodTestRule

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerIndexedFieldsTest.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+		map.put("localized_title", title);
+
+		for (Locale locale : LanguageUtil.getAvailableLocales()) {
+			String languageId = LocaleUtil.toLanguageId(locale);
+
+			String key = "localized_title_" + languageId;
+
+			map.put(key, title);
+			map.put(key.concat("_sortable"), title);
```
Comment (leomiyagi) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerIndexedFieldsTest.java]: Replace this with StringPool.UNDERLINE + Field.SORTABLE_FIELD_SUFFIX

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerIndexedFieldsTest.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+		if (treePaths != null) {
+			if (treePaths.length == 1) {
+				treePath = treePaths[0];
+			}
+			else {
+				treePath = String.valueOf(Arrays.asList(treePaths));
+			}
+		}
+
+		map.put(Field.TREE_PATH, treePath);
```
Comment (leomiyagi) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerIndexedFieldsTest.java]: Is it fine to set this value to null if treePaths is null? Looks like either we don't set the TREE_PATH key value or set with something else.
Whichever the case, the treePath variable is unnecessary. Could do map.put(Field.TREE_PATH, treePaths[0]) at line 222 and do the same at line 225.

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerReindexTest.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ */
+@RunWith(Arquillian.class)
+public class JournalFolderIndexerReindexTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE,
```
Comment (leomiyagi) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/search/test/JournalFolderIndexerReindexTest.java]: This needs to be changed to PermissionCheckerMethodTestRule

**Comment by vagnerbc** [comment]
Changes updated

**Comment by leomiyagi** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by leomiyagi** [comment]
ci:test:search:norebase

**Comment by leomiyagi** [comment]
brandizzi#775

**Pull request closed without merging**